src/helpers/formSubmit.ts

```diff
@@ -1,76 +1,89 @@
 import { SyntheticEvent } from "react";
+import { pipe } from "fp-ts/lib/function";
 import * as E from "fp-ts/lib/Either";
+import * as O from "fp-ts/lib/Option";
 import * as A from "fp-ts/Array";
 
-import type { ValidatorType } from '../components/Field';
+import type { ValidatorType } from '~/components/Field';
 
 type FormEvent = SyntheticEvent<HTMLFormElement>;
-export type FormDataMatrix = [string, string | boolean][];
+export type FormDataItem = [string, FormDataEntryValue];
+export type FormDataMatrix = Array<FormDataItem>;
 export type ErrorsList = Array<{ fieldId: string, message: string }>;
 
-const getFormData = (e: FormEvent): any => {
-    if (!e?.currentTarget) return null;
-    const formData = new FormData(e?.currentTarget);
-    return [...formData.entries()];
+const getFormData = (e: FormEvent): FormDataMatrix | null => {
```

Comment:
I might have some kind of functional programming sickness but it seems to me if it's nullable you can use Option

Reply:
Done at [a9838b6](https://github.com/rafaelpradella/pradella-fp-page/pull/2/commits/a9838b6c72da3cebf6c0ceba36dc61a4623cb8bb)